### DISCUSSION TITLE
Supabase CLI: "Public Schema Does Not Exist"

### DISCUSSION THREAD

**Discussion by masnwilliams** [supabase/supabase#8092] [Questions]
Hello! So I am using Prisma and Supabase together which is working very well for some things and not so well for others.

A problem I recently encountered was when I tried adding the follow supabase database webhook and then running `supabase start`:
```
create trigger "shop-change-listener" 
after insert or update or delete on public.shop 
for each row execute function supabase_functions.http_request(
  'http://localhost:3000/api/v2/hooks/shop',
  'POST',
  '{"Content-type":"application/json"}',
  '1000'
  );
```

This should add a supabase database webhook that sends an HTTP request to that endpoint. However, I get this error, `public.shop does not exist`, and this is because prisma has not yet deployed the migrations which I have set up to run after my `supabase start` command. So the error makes sense but I am looking for a solution.

So I am wondering if there is a way to set up the supabase local development instance's default schema to somehow run the prisma migrate deploy command before running the supabase migrations.

Any help is appreciated!

**Comment by NanoCode012** [comment]
May I ask why are you using both supabase migrate and prisma migrate? Would it be easier to just use one of them?

**Reply by masnwilliams** [reply]
Currently, prisma does not allow the use of multiple schemas nor does it allow access to supabase functions.

We use migrations to automatically create functions for our local development environment as well as an internal distributed test environment.

Prisma works well for database migrations on a single schema and then supabase works for function migrations.

**Reply by NanoCode012** [reply]
Oh I see. Well, an ugly trick that I use to solve a similar problem (`auth` schema) is to move all external schema stuff to a separate sql file/folder and run `psql` on them. It's not intuitive but it works for my small case.

My flow in a package.json script: Prisma Migration -> Run external `psql` to import sql files.

This way, the sql is applied and prisma is still happy..

EDIT: Another method could be to create the empty table in supabase migration. Then inside the prisma migration, alter the table to add the fields inside..

**Reply by masnwilliams** [reply]
Currently, it isn't a big issue because it works great for us. Whenever we need to restart the entire supabase local instance, then we have those supabase migration files run and then when we are just doing migrations, the supabase database is not reset so we don't have to run those again and can just run our Prisma Migrations
